src/lib/allocation_tracker.cc

```diff
@@ -194,6 +190,10 @@ void AllocationTracker::track_allocation(uintptr_t addr, size_t size,
   free_on_consecutive_failures(success);
 
   if (success && _state.track_deallocations) {
+    // \fixme{r1viollet} adjust set to be lock free
```

Comment:
this could probably be replaced with a rather simple sparse bitset over the least significant 48 bits of the pointers, which would allow CAS operations on pages of the bitset rather than a lock.

Reply:
I like this, especially as the impact of collisions are easy to manage. I'll probably make it smaller than that.